### PR TITLE
Define cluster network with cephadm bootstrap

### LINKED ISSUE #139: Wrong cluster network is used
Since commit 16fdedcfe6ffab99c8d8db27df331136bb127d3b, we set the cluster network via `ceph config set mon cluster_network`. This appears to be ignored, resulting in Ceph using the public network for replication.

```
2024-03-18 22:44:15,946 7feee9be7740 INFO Mon IP `10.0.0.1` is in CIDR network `10.0.0.0/24`                                                                                                         
2024-03-18 22:44:15,947 7feee9be7740 INFO Mon IP `10.0.0.1` is in CIDR network `10.0.0.0/24`                                                                                                         
2024-03-18 22:44:15,947 7feee9be7740 DEBUG Inferred mon public CIDR from local network configuration ['10.0.0.0/24', '10.0.0.0/24']                                                                   
2024-03-18 22:44:15,947 7feee9be7740 INFO Internal network (--cluster-network) has not been provided, OSD replication will default to the public_network                                                    
```

### PR DESCRIPTION
**Pull request by priteau**
Fixes #139.

### CODE REVIEW

File: roles/cephadm/tasks/bootstrap.yml [hunk 1]
```diff
@@ -35,19 +36,6 @@
   changed_when: true
   when: not cephadm_check_ceph_conf.stat.exists
 
-- name: Set public network
```
Comment (mnasiadka) [inline roles/cephadm/tasks/bootstrap.yml:38]: That would fix the existing environments, bootstrap is only run when there's no cluster (or you force rebootstrap)

**Review by mnasiadka — APPROVED** [review]

File: roles/cephadm/tasks/bootstrap.yml [hunk 2]
```diff
@@ -43,7 +46,7 @@
 
 - name: Set cluster network
   command:
-    cmd: "cephadm shell -- ceph config set mon cluster_network {{ cephadm_cluster_network }}"
+    cmd: "cephadm shell -- ceph config set global cluster_network {{ cephadm_cluster_network }}"
```
Comment (markgoddard) [inline roles/cephadm/tasks/bootstrap.yml:49]: Doe the public network setting not also need to be global?

Comment (priteau) [reply roles/cephadm/tasks/bootstrap.yml:49]: Actually, when you run `cephadm bootstrap` without any additional `ceph config set`, it sets `public_network` only for `mon`, at least for Quincy v17.2.7: https://github.com/ceph/ceph/blob/v17.2.7/src/cephadm/cephadm#L5452-L5454

```
# ceph config dump | grep public_network
mon           advanced  public_network                          10.0.0.0/24                                                                                     * 
```

Comment (priteau) [reply roles/cephadm/tasks/bootstrap.yml:49]: But Reef does it differently: https://github.com/ceph/ceph/blob/v18.2.2/src/cephadm/cephadm.py#L6009-L6013

Comment (priteau) [reply roles/cephadm/tasks/bootstrap.yml:49]: Let's fix in another change when we understand it better?

Comment (markgoddard) [reply roles/cephadm/tasks/bootstrap.yml:49]: Ack

**Review by markgoddard — APPROVED** [review]

**Pull request merged**